### ISSUE TITLE
tapViewWithAccessibilityLabel fails when accessibilityLabel is not specifically set

### ISSUE THREAD

**Issue by SimonasA** [kif-framework/KIF#596]
AccessibilityLabels works out of the box in iOS: you set title for UIButton and voice over uses that buttons title as a accessibilityLabel if you specifically do not set accesibilityLabel for that UIButton.

I would love to avoid setting accesibilityLabel's for all tested views as it would look more or less like this in all places:

```
[button setTitle:text forState:UIControlStateNormal];
button.accessibilityLabel = text;
```

I mean, - I would love not to dublicate and avoid setting accessibilityLabel for my buttons.

KIF works just fine finding those views with only titles set:
```
[tester waitForViewWithAccessibilityLabel:text];
```

but it fails when it is time to tap those views:
```
[tester tapViewWithAccessibilityLabel:text];
```

The thing is (my guess),- it finds UIButtonLabel (private apple label) as a view with that accessibility label instead of intentional button and fails to tap it.

Is KIF not working with out-of-the box accesibility labels intentionally and by design?
Or that is a bug?

(Sorry if it was already discussed,- checked all open/closed issues,- did not find anything similar)

Thanks

**Comment by jmig** [comment]
For me it works out of the box 99% of the time.

Can you try to use :
````
[tester waitForTappableViewWithAccessibilityLabel:text traits:UIAccessibilityTraitButton];
//And then
[test tapViewWithAccessibilityLabel:text traits:UIAccessibilityTraitButton];
````

**Comment by phatmann** [comment]
@SimonasA, does the solution proposed by @jmig work for you?

**Comment by SimonasA** [comment]
So I updated POD, ran tests again and it works as expected (even without specifically setting ```UIAccessibilityTraitButton```)
I checked what has changed in KIF code and found nothing, so I couldnt say I am fully satisfied :}}
As ir works now, I am closing the issue. 

Guys, thank you very much for your help and attention! ; )

**Comment by Benuuu** [comment]
I'm noticing an issue with this as well, but only specifically on the first launch after a simulator "Resent Contents and Settings".  Looking at the debugger it appears that my UILabels have the text set, but inspecting their accessibilityLabel shows `nil`.

Tested with Xcode 6.3.2 iOS 7.1 simulator against KIF 431a9f11e1e342ec6b27a172742d6e2c15b01532

**Comment by mikelupo** [comment]
What about this commit? Does it still happen there?
https://github.com/kif-framework/KIF/commit/b045a9b195bab5b199a239f0474fb7b350e4c610

**Comment by Benuuu** [comment]
@mikelupo It looks fine on b045a9b195bab5b199a239f0474fb7b350e4c610

I think the offending commit is 927050d0a3e4178182570c560f646fa8a257bc52, but I can't test that on Xcode 6.3.2 iOS 8.1

It's broken on 17a1bc86be31440d144902b1719ae549b173db03, which is the restoration of getting it to work on < iOS 9.

**Comment by Benuuu** [comment]
It looks like with the introduction of 927050d and 17a1bc8, _enableAccessibility is never called.

**Comment by mikelupo** [comment]
Thanks. I've been suspecting one or some of the October commits as I can't use the latest HEAD. 
@thephatmann, what do we do in this case? 

Thanks,
Mike

**Comment by mikelupo** [comment]
Good catch! I'm traveling and cannot review to confirm... but maybe you can submit a PR for a fix?

**Comment by Benuuu** [comment]
@mikelupo I'm testing out a change that will modify + (void)load to be

        if ([XCTestObservationCenter respondsToSelector:@selector(sharedTestObservationCenter)]) {
            XCTestObservationCenter *observationCenter = [XCTestObservationCenter sharedTestObservationCenter];
            [observationCenter addTestObserver:[self sharedAccessibilityEnabler]];
        } else {
            [[self sharedAccessibilityEnabler] _enableAccessibility];
        }

But waiting for local tests to pass on Xcode 6.3.2 iOS 8.1 then Xcode 7.1 iOS 9.1.  This should restore the old behavior, while retaining the new reset behavior for iOS >= 9.

**Comment by mikelupo** [comment]
Thanks again. I won't be able to test the change for myself at HEAD but maybe someone else can. 
Just creating the PR will be a leap forward.






-----Original Message-----

**Comment by readwrite123455** [comment]
tester().waitForView(withAccessibilityLabel: text, traits: UIAccessibilityTraitButton)
        tester().tapView(withAccessibilityLabel: text, traits:UIAccessibilityTraitButton) .  I am getting           caught "NSInvalidArgumentException", "*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil" 
Any one can help me out. Already working from last 7 days .

Any help would be appraciated.
